src/riak_kv_vnode.erl

```diff
@@ -2962,30 +2966,48 @@ perform_put({true, {_Obj, _OldObj}=Objects},
             #putargs{returnbody=RB,
                      bkey=BKey,
                      reqid=ReqID,
+                     coord=Coord,
                      index_specs=IndexSpecs,
                      readrepair=ReadRepair,
-                     coord=Coord}) ->
+                     sync_on_write=SyncOnWrite}) ->
     case ReadRepair of
       true ->
         MaxCheckFlag = no_max_check;
       false ->
         MaxCheckFlag = do_max_check
     end,
-    actual_put(BKey, Objects, IndexSpecs,
-                RB, ReqID, MaxCheckFlag, Coord, State).
+    case SyncOnWrite of
```

Comment:
This way of coding is not Erlang like. In OTP24 you may even get an error or warning.
You bind `Sync` in each branch. Instead you should bind it outside the case with the result of the case.

Reply:
```suggestion
    Sync = case SyncOnWrite of
```